src/components/AddView.js

```diff
@@ -1,10 +1,91 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
+import firebase from 'firebase';
+
+const AddView = ({ token }) => {
+  const [item, setItem] = useState({
+    name: '',
+    howSoon: '7',
+    lastPurchasedDate: null,
+  });
+
+  const handleChange = (e) => {
+    setItem({
+      ...item,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    let docAvailable = false;
+    if (item.name.trim()) {
+      const newItem = {
+        name: item.name,
+        lastPurchasedDate: item.lastPurchasedDate,
+        howSoon: item.howSoon,
+      };
+      db.collection('shoppinglist')
+        .get()
+        .then((querySnapshot) => {
+          querySnapshot.forEach((doc) => {
+            if (doc.data().token === token) {
+              docAvailable = true;
+              doc.ref.update({
+                items: firebase.firestore.FieldValue.arrayUnion(newItem),
+              });
+            }
+          });
+          if (docAvailable === false) {
+            db.collection('shoppinglist').add({
+              items: [newItem],
+              token: token,
+            });
+          }
+        });
+
+      setItem({
+        name: '',
+        howSoon: '7',
+        lastPurchasedDate: null,
+      });
+    } else {
+      alert('Please write an item');
+    }
+  };
 
-const AddView = () => {
   return (
-    <main>
-      <h1>Add Item</h1>
-    </main>
+    <>
+      <form className="form-container" onSubmit={handleSubmit}>
+        <label htmlFor="name">Purchase Item Name:</label>
+        <input
+          id="name"
+          type="text"
+          name="name"
+          value={item.name}
+          onChange={handleChange}
+        />
```

Comment:
[React.doc](https://reactjs.org/docs/forms.html) shows forms with the input inside their label. But I do not know if that is required.